### ISSUE TITLE
Migration file issue when binding to arbitrary model

### ISSUE THREAD

**Issue by rvignesh89** [EliotBerriot/django-dynamic-preferences#96]
I'm just starting to try out the package for my project but ran into an issue. I was able to make it work but wanted to know if this is the recommended approach. My goal was to have preferences for a Company and allow users inside that company to view/edit them. 

#### Environment
macOS Sierra 10.12.6
Python=3.6.1
Django=1.11
djangorestframework=3.6.1

#### Steps

1. Install dynamic_preferences
2. Add only`dynamic_preferences` to INSTALLED_APP before the `Company` app
3. Run `python manage.py migrate` which successfully creates global and user preference models under the dynamic_preferences app.
3. Add following `CompanyPreferenceModel`
```python
class CompanyPreferenceModel(PerInstancePreferenceModel):
    instance = models.ForeignKey(Company)
```
4. Run `python manage.py makemigrations`

After this the migration file is stored into `dynamic_preference` app instead of my app. 

#### Temp fix
By forcing the app_label in my CompanyPreferenceModel I was able to overcome this issue.
```python
class CompanyPreference(PerInstancePreferenceModel):
    instance = models.ForeignKey(Company)

    class Meta:
        app_label = 'my_site'
```

If this is the correct approach for this case then I guess the documentation should be fixed.

**Comment by EliotBerriot** [comment]
Hi and thanks for reporting this. This should work without specifiying an `app_label` but the custom PreferenceModel set up is probably undertested. Would you mind posting the stack trace and your app layout (I'm especially curious to know how your app / project is structured and where your model is located) ?

**Comment by rvignesh89** [comment]
No problem 👍 

I'm not sure what to post for the stack trace since there are no exceptions. Just pasting the makemigrations output here. 

```
Migrations for 'my_site':
  my_site/migrations/0075_market.py
    - Create model Market
Migrations for 'dynamic_preferences':
  /Users/vignesh/.pyenv/versions/my_site/lib/python3.6/site-packages/dynamic_preferences/migrations/0005_auto_20170923_1316.py
    - Create model MarketPreference
    - Alter unique_together for marketpreference (1 constraint(s))
```

As for the code layout I do use a different pattern where I have 1 Django app and all my models in submodules. Something like this.

```
── my_site
      ├── manage.py
      └── my_site
          └── market
              └── models.py
```

**Comment by EliotBerriot** [comment]
Okay, so for some reason the model is not pickud up by Django. I assume you have `my_site.market` under your `INSTALLED_APPS` setting ?

**Comment by czlee** [comment]
If it helps at all, we had to do something similar in our [models.py](https://github.com/czlee/tabbycat/blob/develop/tabbycat/options/models.py) file. It might be an oddity of Meta inheritance, the docs don't say anything about modifying `app_label` on inheritance: https://docs.djangoproject.com/en/1.11/topics/db/models/#meta-inheritance

Maybe there's no special handling for `app_label` so when a subclass inherits its parent's `_meta`, it just blindly copies over the `app_label` too? There doesn't seem to be much documentation on this specific phenomenon, strangely.

**Comment by rvignesh89** [comment]
@EliotBerriot no I don't have `my_site.market` in INSTALLED_APPS. 

@czlee I think you are right. This seems to be a django thing. I tested this using the following class. When I ran `makemigrations` the migration file for `MyUser` is being created in django.contrib.auth app instead of mine. 

```python
@python_2_unicode_compatible
class AbstractBaseUser(models.Model):
    password = models.CharField(_('password'), max_length=128)
    last_login = models.DateTimeField(_('last login'), blank=True, null=True)

    is_active = True

    REQUIRED_FIELDS = []

    class Meta:
        abstract = True
        app_label = 'auth' # Added this into Django's AbstractUser model

class MyUser(AbstractBaseUser):
    pass
```

Which means removing the following from BasePreferenceModel might just do the trick. @EliotBerriot is there any purpose of specifying the `app_label` here. If this didn't cause any issues before then maybe some recent Django upgrade did this?

```python
class BasePreferenceModel(models.Model):

    """
    A base model with common logic for all preferences models.
    """

    ...

    class Meta:
        abstract = True
        # app_label = 'dynamic_preferences'

```

**Comment by EliotBerriot** [comment]
I think this is strange, as I have successfully subclassed the `PreferenceModel` at work in the past (proprietary code that I cannot link to, unfortunately), without explicitely setting the `app_label`. I got the corresponding migration generated proprely.

Maybe it's working differently on various django versions? Until we're not sure, I'd rather not simply comment the line: I'm afraid it could silently introduce backward incompatible changes in some setups.

@rvignesh89 I cannot remember why I introduced this in the first place. A `git  blame` shows it was introduced in 75dadd50, three yers ago, but it's mixed with a lot of other changes from that time. It was early development then, maybe it was as a hack.

**Comment by rvignesh89** [comment]
Understood. I also feel that commenting this out might cause subtle issues for those already using the app. How about if we add a note on the docs for future users of this package? Because if they are following the steps written there I'm pretty sure they'll run into this problem.

**Comment by EliotBerriot** [comment]
Yep, it makes sense to upgrade the documentation to reflect this. I'll work on that tomorrow :)

**Comment by EliotBerriot** [comment]
I submitted a documentation PR, but fixing this properly will porobably imply breaking backward compatibility. So I've just assigned it to a 2.0 milestone release.
